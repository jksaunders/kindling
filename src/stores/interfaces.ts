import { AuthStore } from './AuthStore';
import { UserStore } from './UserStore';

export interface BaseStore {
  stores: Stores;
  setStores: (stores: Stores) => void;
}

export interface Stores {
  authStore: AuthStore;
  userStore: UserStore;
}
