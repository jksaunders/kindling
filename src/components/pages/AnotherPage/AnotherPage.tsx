import { FC } from 'react';
import styled from 'styled-components';
import { Button } from '@jksaunders/bonfire';

const StyledComponentsExample = styled.div`
  border-radius: 10px;
  box-shadow: 0px 4px 15px rgba(0, 0, 0, 0.1);
`;

const AnotherPage: FC = () => (
  <StyledComponentsExample>
    I&apos;m another example page!
    <Button color="blue" />
  </StyledComponentsExample>
);

export default AnotherPage;
