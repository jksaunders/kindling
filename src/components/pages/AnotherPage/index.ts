import AnotherPage from './AnotherPage';

export default AnotherPage;
