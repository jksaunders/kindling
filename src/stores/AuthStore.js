import { useLocalStore } from 'mobx-react-lite';

const useAuthStore = () =>
  useLocalStore(() => ({
    stores: null,
    setStores(stores) {
      this.stores = stores;
    },
    isLoggedIn: false,
  }));

export default useAuthStore;
