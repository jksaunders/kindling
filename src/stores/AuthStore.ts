import { useLocalObservable } from 'mobx-react-lite';
import { BaseStore, Stores } from './interfaces';

export interface AuthStore {
  isLoggedIn: boolean;
}

const useAuthStore = (): AuthStore =>
  useLocalObservable((): BaseStore & AuthStore => ({
    stores: {} as Stores,
    setStores(stores): void {
      this.stores = stores;
    },
    isLoggedIn: false,
  }));

export default useAuthStore;
