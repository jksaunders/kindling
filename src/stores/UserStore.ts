import { useLocalObservable } from 'mobx-react-lite';
import { BaseStore, Stores } from './interfaces';

export interface UserStore {
  name: string;
}

const useUserStore = (): UserStore =>
  useLocalObservable((): BaseStore & UserStore => ({
    stores: {} as Stores,
    setStores(stores): void {
      this.stores = stores;
    },
    name: 'User store name',
  }));

export default useUserStore;
