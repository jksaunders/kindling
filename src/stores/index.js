import React from 'react';
import useAuthStore from './AuthStore';

const LocalStoresContext = React.createContext();

// eslint-disable-next-line react/prop-types
export const LocalStoresRoot = ({ children }) => {
  const stores = {
    authStore: useAuthStore(),
  };

  Object.keys(stores).forEach((key) => {
    if (typeof stores[key].setStores === 'function') {
      stores[key].setStores(stores);
    }
  });

  return (
    <LocalStoresContext.Provider value={stores}>
      {children}
    </LocalStoresContext.Provider>
  );
};

const useStores = () => React.useContext(LocalStoresContext);

export default useStores;
