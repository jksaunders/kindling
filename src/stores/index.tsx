import React, { FC } from 'react';
import { Stores } from './interfaces';
import useAuthStore from './AuthStore';
import useUserStore from './UserStore';

// `as Stores` to override compile errors
const LocalStoresContext = React.createContext<Stores>({} as Stores);

export const LocalStoresRoot: FC = ({ children }) => {
  const stores: Stores = {
    authStore: useAuthStore(),
    userStore: useUserStore(),
  };

  Object.keys(stores).forEach((key) => {
    if (typeof stores[key].setStores === 'function') {
      stores[key].setStores(stores);
    }
  });

  return (
    <LocalStoresContext.Provider value={stores}>
      {children}
    </LocalStoresContext.Provider>
  );
};

const useStores = (): Stores => React.useContext(LocalStoresContext);

export default useStores;
