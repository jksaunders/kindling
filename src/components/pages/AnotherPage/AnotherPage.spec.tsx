import React from 'react';
import { render, screen } from '@testing-library/react';
import AnotherPage from './AnotherPage';
import { LocalStoresRoot } from '../../../stores';

test('renders', async () => {
  const {
    container: { firstChild },
  } = render(
    <LocalStoresRoot>
      <AnotherPage />
    </LocalStoresRoot>
  );
  await screen.findByText("I'm another example page!");
  expect(screen.getByText("I'm another example page!")).toBeInTheDocument();
  expect(screen.getByText("I'm another example page!")).toHaveStyle(
    `border-radius: 10px`
  );
  expect(firstChild).toHaveStyle(`border-radius: 10px`);
  expect(firstChild).toMatchSnapshot();
});
