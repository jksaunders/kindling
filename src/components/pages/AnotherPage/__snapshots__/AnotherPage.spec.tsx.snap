// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c1 {
  background-color: blue;
}

.c0 {
  border-radius: 10px;
  box-shadow: 0px 4px 15px rgba(0,0,0,0.1);
}

<div
  class="c0"
>
  I'm another example page!
  <button
    class="c1"
    color="blue"
  >
    test button
  </button>
</div>
`;
